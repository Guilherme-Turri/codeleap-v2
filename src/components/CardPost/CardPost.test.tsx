import { pageRender } from '../../render/pageRender'
import { CardPost } from '.'
import { screen } from '@testing-library/react'
describe('<CardPost />', () => {
  const cardProps = {
    author: 'AuthorTest',
    title: 'This is a Tittle',
    content: 'This is the content',
    _id: '123',
    authorId: '123',
    createdAt: '2022-09-23T14:25:37.504+00:00',
    user: '123'
  }
  it('should render on screen properly', () => {
    const { container } = pageRender(<CardPost {...cardProps} />)
    const title = screen.getByText('This is a Tittle');
    const content = screen.getByText('This is the content');
    expect(title).toBeInTheDocument();
    expect(content).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  })
  it('should render properly on a mobile', () => {
    window.innerWidth = 400;
    pageRender(<CardPost {...cardProps} />)
    const cardContainer = screen.getByTestId('cardContainer');
    expect(cardContainer).toHaveStyleRule('flex-direction', 'column')
  })

}

)