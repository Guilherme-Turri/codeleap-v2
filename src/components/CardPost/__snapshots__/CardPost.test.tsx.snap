// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardPost /> should render on screen properly 1`] = `
.c0 {
  min-height: 150px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 0 0 85%;
  -ms-flex: 0 0 85%;
  flex: 0 0 85%;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  margin-top: 0.313rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 0 20%;
  -ms-flex: 0 20%;
  flex: 0 20%;
}

.c1 a {
  color: #c4d299;
  font-weight: bold;
  cursor: pointer;
}

.c2 {
  width: 5.625rem;
  height: 5.625rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 50%;
  object-fit: cover;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 80%;
  -ms-flex: 0 80%;
  flex: 0 80%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  padding: 0.625rem;
}

.c3 > *:nth-child(1) {
  color: #c4d299;
  font-weight: bold;
}

.c3 > *:nth-child(3) {
  font-size: 0.85rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div>
  <main
    class="c0"
    data-testid="cardContainer"
  >
    <div
      class="c1"
    >
      <img
        alt="Profile"
        class="c2"
      />
      <a
        href="/user/123"
      >
        @
        AuthorTest
      </a>
    </div>
    <div
      class="c3"
    >
      <p
        class="c4"
      >
        This is a Tittle
      </p>
      <p
        class="c4"
      >
        This is the content
      </p>
      <p
        class="c4"
      >
        7 months ago
      </p>
    </div>
  </main>
</div>
`;
